### PR TITLE
docs: add limit-lep plug blog

### PR DESCRIPTION
**Pull request by SylviaBABY**

### CODE REVIEW

**Review by yzeng25 — APPROVED** [review]

File: _posts/blog/zh-CN/limit-rep-plug.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+Apache APISIX 是一个生产可用的七层全流量处理平台，可作为 API 网关处理业务入口流量，具有极高性能、超低延迟的显著特性。它内置了 50 多种插件，覆盖身份验证、安全防护、流量控制、Serverless、可观测性等多个方面，可满足企业客户常见的使用场景。
+
+如下方架构图所示，Apache APISIX 分为数据面（左侧）与控制面（右侧）两部分：通过控制面下发配置到 ETCD，数据面借助丰富的插件处理内外流量。
+
+![Apache APISIX 架构图](https://static.apiseven.com/202108/1630656070351-bc49128b-e0ef-4c93-830c-2ca7931dacce.png)
+
+Apache APISIX 暴露了一组接口，方便我们为 API 绑定插件。如果我们希望为 API 增加限速能力，只需为 API 绑定 `limit-req` 插件：
+
+```
```
Comment (yzeng25) [inline _posts/blog/zh-CN/limit-rep-plug.md:21]: "```shell" or

 "```json" ? cc @Serendipity96

File: _posts/blog/zh-CN/limit-rep-plug.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+那么 Apache APISIX 是如何与低代码能力结合的呢？这需要数据面 Apache APISIX 与控制面 Apache APISIX Dashboard 共同配合完成。整体流程如下：
+
+![低代码能力实现原理](https://static.apiseven.com/202108/1630657206662-02f0f699-8b9d-40f0-b36b-dbf700b16e02.png)
+
+### Apache APISIX
+
+在 Apache APISIX 中，我们在 Route 实体中新增了 [script 执行逻辑](https://github.com/apache/apisix/pull/1982)，可用于接收 Dashboard 生成的 Lua 函数并执行，它支持调用已有插件以复用代码。另外，它也作用于 HTTP 请求的生命周期中的各个阶段，如 `access`、`header_filer`、`body_filter` 等，系统会在相应阶段自动执行 `script` 函数对应阶段代码，见如下 `script` 示例：
+
+```
```
Comment (yzeng25) [inline _posts/blog/zh-CN/limit-rep-plug.md:73]: ditto cc @Serendipity96

File: _posts/blog/zh-CN/limit-rep-plug.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+### Apache APISIX Dashboard
+
+在 Dashboard 中，它包含了 Web 与 ManagerAPI 共两个子组件：Web 用于提供可视化界面，方便我们配置 API 网关；ManagerAPI 用于提供 RESTful API，供 Web 或其它客户端调用以便操作配置中心（默认为 ETCD），进而间接地控制 Apache APISIX。
+
+为了生成合法、有效的 script 函数，ManagerAPI 选择了 DAG 有向无环图的数据结构进行底层设计，并自主研发了 [`dag-to-lua`](https://github.com/api7/dag-to-lua) 项目：它将根节点作为开始节点，根据判断条件决定下一个流转插件，这将有效避免逻辑死循环。如下为 DAG 数据结构的示意图：
+
+![DAG 数据结构示意图](https://static.apiseven.com/202108/1630658809809-af6b83e5-af52-4ad8-b9df-d406156f58ec.png)
+
+对应到 ManagerAPI 接收的 `script` 参数上，示例如下：
+```
```
Comment (yzeng25) [inline _posts/blog/zh-CN/limit-rep-plug.md:87]: 1. add a new line between 86 and 87
2. ditto @Serendipity96

File: _posts/blog/zh-CN/limit-rep-plug.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+## 未来展望
+
+通过拖拽的方式，可以使得使用人员更方便地组合插件来满足不同的场景，以提升 API 网关可扩展能力与运维体验。在实际使用过程中，存在如下可以继续优化的问题：
+
+1. 目前元件的边界判断条件还不够丰富，通过继续完善这些条件，以减少不合理的编排组合；
+2. 当前编排示例不多，提供更多的参考示例可方便开发者学习、供用户使用；
+3. 当前 Apache APISIX 使用了插件定义的 code 进行状态返回（异常则返回状态码，请求终止），可以支持更多 HTTP Response 字段甚至修改插件定义来扩展插件编排能力，如下述插件定义：
+
+```
```
Comment (yzeng25) [inline _posts/blog/zh-CN/limit-rep-plug.md:155]: ditto cc @Serendipity96

**Review by yzeng25 — CHANGES_REQUESTED** [review]

File: _posts/blog/zh-CN/limit-rep-plug.md (hunk 1 above)
Comment (Serendipity96) [reply _posts/blog/zh-CN/limit-rep-plug.md:21]: yeah  we need add type

Comment (SylviaBABY) [reply _posts/blog/zh-CN/limit-rep-plug.md]: done

File: _posts/blog/zh-CN/limit-rep-plug.md (hunk 3 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/limit-rep-plug.md]: done

File: _posts/blog/zh-CN/limit-rep-plug.md (hunk 2 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/limit-rep-plug.md]: done

File: _posts/blog/zh-CN/limit-rep-plug.md (hunk 4 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/limit-rep-plug.md]: done

**Review by yzeng25 — APPROVED** [review]

**Pull request merged**